internal/pkg/gateway/commit/notifier_test.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2021 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit_test
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+	commonProto "github.com/hyperledger/fabric-protos-go/common"
+	peerProto "github.com/hyperledger/fabric-protos-go/peer"
+	deliverMock "github.com/hyperledger/fabric/common/deliver/mock"
+	"github.com/hyperledger/fabric/internal/pkg/gateway/commit"
+	commitMocks "github.com/hyperledger/fabric/internal/pkg/gateway/commit/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNotifier(t *testing.T) {
+	type Transaction struct {
+		ID     string
+		Status peerProto.TxValidationCode
+	}
+
+	NewPayload := func(transactionID string) *commonProto.Payload {
+		channelHeader := &commonProto.ChannelHeader{
+			Type: int32(commonProto.HeaderType_ENDORSER_TRANSACTION),
+			TxId: transactionID,
+		}
+		channelHeaderBytes, err := proto.Marshal(channelHeader)
+		assert.NoError(t, err)
```

Comment:
Here, and in the rest of these setup function, we'd be better off with `require.NoError`, since if this step fails, almost certainly we will encounter future red herrings in assertions.

Reply:
👍